include/NAS2D/Renderer/Point.h

```diff
@@ -12,79 +12,97 @@
 
 namespace NAS2D {
 
-struct Point_2d;
-struct Point_2df;
+
+template <typename BaseType>
+struct Point {
+	Point() = default;
+	Point(const Point& point) = default;
+	Point& operator=(const Point& other) = default;
+
```

Comment:
Consider `=default`ing the move operations and the Dtor to adhere to Rule of Five/Six and [C.21](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-five).

It's not necessarily a requirement (and I'm still not sure if moves of POD types aren't silently converted to copies anyway) but may be a missed opportunity for compiler optimization.